### PR TITLE
Remove entryPoint if not required

### PR DESCRIPTION
**Pull request by beaufortfrancois**
See https://github.com/gpuweb/gpuweb/pull/4387/

⚠️ Do not merge this PR until Chrome 121 reaches stable channel.

### CODE REVIEW

**Comment by Kangz** [comment]
FYI @jimblandy @mwyrzykowski this will simplify samples a bit but will require entryPoint name defaulting.

**Review by mwyrzykowski — APPROVED** [review]

**Comment by kainino0x** [comment]
Converting to draft to make the PR harder to accidentally merge.

**Comment by beaufortfrancois** [comment]
We can now merge it as Chrome 121 is stable.

**Review by austinEng — APPROVED** [review]

**Review by mwyrzykowski — APPROVED** [review]

**Comment by greggman** [comment]
ps: Firefox still requires entryPoint as of nightly 124.0a1 (2024-01-25) (64-bit)

@kdashg @jimblandy

**Review by kdashg — CHANGES_REQUESTED** [review]
Just a sec!

**Comment by jimblandy** [comment]
So, we agree this is a sweet readability hack, and would like to see it deployed in the examples - but since it will break Firefox, we would like to wait to merge this PR until https://github.com/gfx-rs/wgpu/issues/5145 is closed.

**Comment by jimblandy** [comment]
Note that we need both https://github.com/gfx-rs/wgpu/issues/5145 (the `wgpu`-side work) and Bugzilla [Bug 1877488](https://bugzilla.mozilla.org/show_bug.cgi?id=1877488) (the Firefox bindings work) fixed before Firefox will be able to handle omitted entry point names. @erichdongubler is actively working on these.

**Comment by beaufortfrancois** [comment]
Firefox Nightly now handles omitted entry point names (I've checked locally). 
Since Safari Tech Preview and Chrome also handle them properly, I believe we can now merge this PR.

Note that I've just rebased it due to merging conflicts.

**Comment by ErichDonGubler** [comment]
As the person who worked on this support in Firefox, I'm excited to see this land. 😁

**Review by Kangz — APPROVED** [review]

**Comment by beaufortfrancois** [comment]
I've had to update the PR again. Please review and merge ;)

File: package-lock.json
Comment (ben-clayton) [inline package-lock.json:1]: Are the changes to `package-lock.json` intentional?

Comment (beaufortfrancois) [reply package-lock.json:1]: Yes. They are due to me running `npm i` because of iframe changes.

Comment (beaufortfrancois) [reply package-lock.json:1]: You're right. I'll revert package-lock.json changes then to not pollute this PR.

Comment (beaufortfrancois) [reply package-lock.json:1]: Done

**Comment by beaufortfrancois** [comment]
@kainino0x Can you merge?

**Comment by greggman** [comment]
I'll merge

**Pull request merged**